src/dash-table/components/FilterFactory.tsx

```diff
@@ -65,26 +65,26 @@ export default class FilterFactory {
 
         const rawGlobalFilter = R.map(
             ast => ast.query || '',
-            R.filter(ast => Boolean(ast), asts)
+            R.filter<SingleColumnSyntaxTree>(ast => Boolean(ast), asts)
         ).join(' && ');
```

Comment:
not new in this PR, but what's `ast => ast.query || ''` about? If we actually ever got `''` we'd see `' &&  && '` in the result, wouldn't we?

Reply:
🐑 You're right. Will try and figure out why I did this in the first place and replace it..

Reply:
Ok, so here I am creating a `raw` query for caching purposes. It does not matter that the resulting query could contain `&& &&` or any other invalid "real" query combination. I am simply making sure the top-level table component has something to hang on to -- to know the user changed something in order to re-render / bust caching on the appropriate parts (e.g. filter validation / highlighting). This could probably be reworked / improved so as to not require this made up field in the future.

Otherwise, one could argue that I could filter out empty queries but that's about it. On the other hand, this other usage [here](https://github.com/plotly/dash-table/pull/410/files#diff-cb425151ed75b9da3b79ada8820b03aeR15) did need to be changed for the same reasons you mentioned above.